### PR TITLE
Add documentation/comments for PanelPluginsModel

### PR DESCRIPTION
**Pull request by DrTobe**
1. Mainly document the header file.
2. Explain how to calculate the right indexes in movePlugin.

If there is anything missing, difficult to understand or too verbose, please let me know.

### CODE REVIEW

**Comment by palinek** [comment]
IMO the specific usage of this model (from configure, plugin move by mouse) should be just as notes. All in all it is a model and should not restrict its usage to one specific purpose only.

**Comment by DrTobe** [comment]
Do you mean that I should omit comments like "This method should be called when the user moved a Plugin in the config dialog so that this change will be reflected in the data." because it is too specific to the current usage of the model?

In general, I agree and the model description should be independent from its usage. As it is used only once in the whole application I thought this kind of documentation could help understand how everything is connected.

**Comment by palinek** [comment]
> As it is used only once in the whole application I thought this kind of documentation could help understand how everything is connected.

Yep... therefore I've written, that those statements should be marked as *\\note*

P.S.: just curious, why do you prefer the JavaDoc style marks over the DoxyGen style?

**Comment by DrTobe** [comment]
Yes, I like this proposal, too. Then, we have separated the 'real' documentation for the class and the additional documentation that helps understanding the whole application.

Concerning the JavaDoc style / DoxyGen style, I am using Qt Creator and whenever I start a comment, it automatically adds '@brief' and maybe some other Doxygen commands. Is there any preference in LXQt? For sure, it wouldn't be a problem to change that.

**Comment by palinek** [comment]
> Concerning the JavaDoc style / DoxyGen style, I am using Qt Creator and whenever I start a comment, it automatically adds '@brief'

I'm not a regular user of Creator, but just tested it. If you start as the JavaDoc style `/**`, it adds `@brief`. If you start with the DoxyGen/Qt style `/*!`, it adds `\brief`

> Is there any preference in LXQt?

I don't know... just that we are based on Qt, I tend to say we should prefer the DoxyGen/Qt style.

@jleclanche, @PCMan, @luis-pereira, @paulolieuthier is there any preference?

**Comment by paulolieuthier** [comment]
I don't know if there is something official, but we do tend to follow Qt and KDE.

**Comment by DrTobe** [comment]
Ok, that's reasonable. I'll change that. And in another commit / pull-request, I will change the other classes, too, so that it gets consistent.

**Comment by luis-pereira** [comment]
I don't remember any policy or guidelines being set.

**Comment by jleclanche** [comment]
There's no official policy/guideline yeah. Like @paulolieuthier we follow qt/kde.

**Comment by DrTobe** [comment]
1. Marked usage-specific information with \note
2. Changed to DoxyGen documentation style

**Comment by palinek** [comment]
GTM

File: panel/panelpluginsmodel.h [hunk 1]
```diff
@@ -48,52 +55,284 @@ class PanelPluginsModel : public QAbstractListModel
...
+     *
+     * To access the elements, you can use indexing or an iterator on the
+     * list. For each element p, p.first is the name of the Plugin as it
+     * is used in the configuration files, p.second.data() is the Plugin.
+     *
+     * \sa pluginslist_t
+     */
     pluginslist_t mPlugins;
+    /*!
+     * \brief mPanel Stores a reference to the LXQtPanel.
```
Comment (paulolieuthier) [inline panel/panelpluginsmodel.h]: I'm not sure it's correct to say this. `mPanel` is a pointer to the object, does it store a *reference*? Anyway, I think the comments (although very valuable) make more sense for the functions, so the member objects should be left alone (unless one really needs explanation). Besides, in some cases (like this one), the comment is very obvious, and so it pollutes more than it helps.

Don't get me wrong, you're doing a great job. Besides all I said, GTM.

**Comment by palinek** [comment]
landed in @9a1225b, thanks

**Pull request closed without merging**